src/request.cpp

```diff
@@ -191,11 +247,12 @@ bool Request::empty() const
 
 void Request::swap(Request& r)
 {
-    // we use swap to move between queues, but process only after it gets into the completedreqs
+    // we use swap to move between queues, but process only after it gets completed
     cmds.swap(r.cmds);
     assert(jsonresponse.empty() && r.jsonresponse.empty());
     assert(json.pos == NULL && r.json.pos == NULL);
     assert(processindex == 0 && r.processindex == 0);
+    std::swap(mV3, r.mV3);
```

Comment:
I found this call to swap confusing. Isn't it simply doing `mV3 = r.mV3`?
Also, and this one is just a personal preference, I'd add this line just below the `cmds.swap(r.cmds)` :smiley:

Reply:
I agree it would be better with those lines together - maybe the asserts got added after.   It's not a simple assignment, it really does swap the values between the two variables.   As this is Request::swap(), probably it should be swapping every member of Request - though the others are not in use at this stage and will all be zero/empty.  But for simplicity, an objects's swap function should swap all members.

Reply:
I'll include a small update for this in the PR, with comments.   The asserts check we are not using the fields that are not swapped